controller/state.go

```diff
@@ -1,29 +1,67 @@
 package controller
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/filecoin-project/dealbot/controller/client"
 	"github.com/filecoin-project/dealbot/tasks"
 	"github.com/pborman/uuid"
 )
 
-var Tasks []*tasks.Task
+var State *state
+
+type state struct {
+	tasks []*tasks.Task
+	mu    sync.Mutex
+}
+
+func (s *state) MarshalJSON() ([]byte, error) {
+	return json.Marshal(s.tasks)
+}
+
+func (s *state) Update(req *client.UpdateTaskRequest) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	for _, t := range s.tasks {
+		if t.UUID == req.UUID {
+			if t.Status == tasks.Available {
+				log.Infow("state update", "uuid", t.UUID, "status", req.Status, "worked_by", req.WorkedBy)
```

Comment:
This needs a lot more improvement. This is just the basic `update` from `Available` to `InProgress` - we will need something much more elaborate to properly track Tasks and updates.